neuralop/datasets/data_transforms.py

```diff
@@ -64,6 +64,138 @@ def forward(self, **data_dict):
+        self.current_res = self.sub_to_res(self.current_sub)           
+        
+        print(f'Original Incre Res: change index to {self.current_index}')
+        print(f'Original Incre Res: change sub to {self.current_sub}')
+        print(f'Original Incre Res: change res to {self.current_res}')
+            
+    def wrap(self, model):
+        self.model = model
+        return self
+
+    def to(self, device):
+        if self.in_normalizer is not None:
+            self.in_normalizer = self.in_normalizer.to(device)
+        if self.out_normalizer is not None:
+            self.out_normalizer = self.out_normalizer.to(device)
+        self.device = device
+        return self
+
+    def sub_to_res(self, sub):
+        # Convert sub to resolution based
+        return int(self.dataset_resolution / sub)
+
+    def epoch_wise_res_increase(self, epoch):
+        # Update the current_sub and current_res values based on the epoch
+        if epoch % self.epoch_gap == 0 and epoch != 0 and (
+                self.current_logged_epoch != epoch):
+            self.current_index += 1
+            self.current_sub = self.index_to_sub_from_table(self.current_index)
+            self.current_res = self.sub_to_res(self.current_sub)
+            self.current_logged_epoch = epoch
+
+            if self.verbose:
+                print(f'Incre Res Update: change index to {self.current_index}')
+                print(f'Incre Res Update: change sub to {self.current_sub}')
+                print(f'Incre Res Update: change res to {self.current_res}')
+
+    def index_to_sub_from_table(self, index):
+        # Get the sub value from the sub_list based on the index
+        if index >= len(self.sub_list):
+            return self.sub_list[-1]
+        else:
+            return self.sub_list[index]
+
+    def regularize_input_res(self, x, y):
+        # Regularize the input data based on the current_sub and dataset_name
+        for idx in self.dataset_indices:
+            indexes = torch.arange(0, x.size(idx), self.current_sub, device=self.device)
+            x = x.index_select(dim=idx, index=indexes)
+            y = y.index_select(dim=idx, index=indexes)
+        return x, y
+    
+    def step(self, loss=None, epoch=None, x=None, y=None):
+        if x is not None and y is not None:
+            self.epoch_wise_res_increase(epoch)
+            return self.regularize_input_res(x, y)
+        
+    def preprocess(self, data_dict, epoch=None, mode = "Train", batched=True):
+        x = data_dict['x'].to(self.device)
+        y = data_dict['y'].to(self.device)
+
+        if self.in_normalizer is not None:
+            x = self.in_normalizer.transform(x)
+        if self.positional_encoding is not None:
+            x = self.positional_encoding(x, batched=batched)
+        if self.out_normalizer is not None and self.train:
+            y = self.out_normalizer.transform(y)
+        
+        if mode == "Train":
```

Comment:
That makes sense, but the preprocess is where we are actually changing the samples, so that requires info of what mode we are using (the mode is actually changed in on_train_batch_start and on_val_batch_start).